packages/server/graphql/mutations/helpers/notifications/SlackNotifier.ts

```diff
@@ -297,5 +297,121 @@ export const SlackNotifier: Notifier = {
 
   async integrationUpdated() {
     // Slack sends a system message on its own
+  },
+
+  async shareTopic(
+    dataLoader: DataLoaderWorker,
+    userId: string,
+    teamId: string,
+    meetingId: string,
+    reflectionGroupId: string,
+    stageIndex: number,
+    channelId: string
+  ) {
+    const r = await getRethink()
+    const [user, team, meeting, reflectionGroup, reflections, slackAuth] = await Promise.all([
+      dataLoader.get('users').load(userId),
+      dataLoader.get('teams').load(teamId),
+      dataLoader.get('newMeetings').load(meetingId),
+      dataLoader.get('retroReflectionGroups').load(reflectionGroupId),
+      r.table('RetroReflection').getAll(reflectionGroupId, {index: 'reflectionGroupId'}).run(),
+      r.table('SlackAuth').getAll(userId, {index: 'userId'}).filter({teamId})(0).run()
+    ])
+
+    if (!user) {
+      throw new Error('User not found')
+    }
+
+    if (!team) {
+      throw new Error('Team not found')
+    }
+
+    if (!slackAuth) {
+      throw new Error('Slack auth not found')
+    }
+
+    const {botAccessToken, slackUserId} = slackAuth
+    const manager = new SlackServerManager(botAccessToken!)
+    const channelInfo = await manager.getConversationInfo(channelId)
+
+    if (channelId !== slackUserId) {
+      if (!channelInfo.ok) {
+        throw new Error(channelInfo.error)
+      }
+      const {channel} = channelInfo
+      const {id: channelId, is_member: isMember, is_archived: isArchived} = channel
+      if (isArchived) {
+        throw new Error('Slack channel archived')
+      }
+      if (!isMember) {
+        const joinConvoRes = await manager.joinConversation(channelId)
+        if (!joinConvoRes.ok) {
+          throw new Error('Unable to join slack channel')
+        }
+      }
+    }
+
+    const topic = reflectionGroup.title
+    const discussionUrl = makeAppURL(appOrigin, `meet/${meetingId}/discuss/${stageIndex + 1}`)
+    const meetingUrl = makeAppURL(appOrigin, `meet/${meetingId}`)
+
+    const reflectionsText = reflections
+      .map((reflection) => `• ${reflection.plaintextContent}`)
+      .join('\n')
+
+    const slackBlocks = [
+      {
+        type: 'section',
+        text: {
+          type: 'mrkdwn',
+          text: `<@${slackAuth.slackUserId}> has shared reflections about *"${topic}"* from their retrospective`
+        }
+      },
+      {
+        type: 'section',
```

Comment:
+1 you can use the `makeSection` function here to reduce the code: https://github.com/ParabolInc/parabol/blob/88c2bd5e41cce781ca497a6047642f2c46e73680/packages/server/graphql/mutations/helpers/notifications/makeSlackBlocks.ts